src/main/java/ch/njol/skript/test/runner/SkriptJUnitTest.java

```diff
@@ -0,0 +1,99 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.test.runner;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Pig;
+import org.junit.After;
+
+/**
+ * Class that helps the JUnit test communicate with Skript.
+ */
+public abstract class SkriptJUnitTest {
+
+	/**
+	 * Used for getting the last ran JUnit test name.
+	 */
+	public static String lastJUnitTest;
+
+	private static long d;
+
+	/**
+	 * The delay this JUnit test is requiring to run.
+	 * 
+	 * @return the delay in milliseconds this junit test is requiring to run for.
+	 */
+	public static long getDelay() {
+		return d;
+	}
+
+	/**
+	 * @param delay Add a delay in milliseconds for this test to run.
+	 */
+	public static void setDelay(long delay) {
+		d = delay;
+	}
```

Comment:
Why does this exist? Tests may take a different amount of time on every machine.

Reply:
The server needs to stay open to run the tests, so the test needs to provide an estimated time to keep that said test open for. The test runner will close the server after the tests are complete and move on to the next test environment.

Reply:
Can't the tests notify when they're done?